### PR TITLE
Add bootstrap storage account to genesis

### PR DESCRIPTION
**Pull request by sagar-solana**
#### Problem

Storage account creation consumed the last remaining token in the bootstrap leader's account. This resulted in the node being removed from leader_rotation/stake holders. 

#### Summary of Changes

Add passive staking for storage accounts

### CODE REVIEW

**Comment by sagar-solana** [comment]
@rob-solana do I need anything else for validators? It seems like validators should have enough tokens (fullnode.sh gives them stake*2+1)

**Comment by mvines** [comment]
Does `run.sh` need this too?

**Comment by sagar-solana** [comment]
> Does `run.sh` need this too?

Yes it does! Thanks!

File: genesis/src/main.rs [hunk 1]
```diff
@@ -201,6 +213,11 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                     bootstrap_leader_stake_lamports,
                 ),
             ),
+            // storage account
+            (
+                bootstrap_storage_keypair.pubkey(),
+                Account::new(1, 1024 * 4, &system_program::id()),
```
Comment (rob-solana) [inline genesis/src/main.rs]: please drive this with a parameter

**Comment by rob-solana** [comment]
> @rob-solana do I need anything else for validators? It seems like validators should have enough tokens (fullnode.sh gives them stake*2+1)

stake*2+1 because one token stays with the node account, and we put stakes in both the vote account and stake account (migration)

ultimately, the vote account needs just one token, so it'd be stake + 2.

File: genesis/src/main.rs (hunk 1 above)
Comment (sagar-solana) [reply genesis/src/main.rs]: Which part?

Comment (mvines) [reply genesis/src/main.rs]: `1024 * 4`

Comment (sagar-solana) [reply genesis/src/main.rs]: I figured it was for whenever we address that todo.

Comment (mvines) [reply genesis/src/main.rs]: yeah but now that `1024 * 4` is split across 3 random files.  Time to take the bullet I think :)

Comment (sagar-solana) [reply genesis/src/main.rs]: In this PR? master is broken-ish.

Comment (sagar-solana) [reply genesis/src/main.rs]: Can I follow up with another PR for that @rob-solana ? Will be afk for most of this evening.

Comment (mvines) [reply genesis/src/main.rs]: wfm

File: genesis/src/main.rs [hunk 2]
```diff
@@ -97,6 +97,15 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                 .required(true)
                 .help("Path to file containing the bootstrap leader's staking keypair"),
         )
+        .arg(
+            Arg::with_name("bootstrap_storage_keypair_file")
+                .short("r")
```
Comment (mvines) [inline genesis/src/main.rs]: actually please just remove the short option.  It's not really helpful

Comment (sagar-solana) [reply genesis/src/main.rs]: Will do. (although I was thinking of it as the "repliaction" account"  :man_shrugging:)

File: genesis/src/main.rs [hunk 3]
```diff
@@ -201,6 +212,11 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                     bootstrap_leader_stake_lamports,
                 ),
             ),
+            // storage account
+            (
+                bootstrap_storage_keypair.pubkey(),
+                Account::new(1, 1024 * 4, &solana_storage_api::id()),
```
Comment (rob-solana) [inline genesis/src/main.rs]: did this require a change to Cargo.toml?

Comment (sagar-solana) [reply genesis/src/main.rs]: Nope, got through CI fine

Comment (rob-solana) [reply genesis/src/main.rs]: what's the 4k size thing?  can you call storage_api to construct this account for you, instead?

**Comment by rob-solana** [comment]
> @rob-solana do I need anything else for validators? It seems like validators should have enough tokens (fullnode.sh gives them stake*2+1)

looks like you've got this handled already

File: genesis/src/main.rs (hunk 3 above)
Comment (sagar-solana) [reply genesis/src/main.rs]: That's the data buffer size for the account. Yeah I'm planning to follow up with a PR that removes all uses of the magic numher.

**Pull request merged**